### DISCUSSION TITLE
Response delay percentile (P25, P50, P75, P85, P95, etc.) statistical function

### DISCUSSION THREAD

**Discussion by peoplewhoworkhard** [prometheus/prometheus#12601] [Q&A]
### Proposal

Hello, I would like to ask whether prometheus has the function of calculating the response delay percentile (pxx), and if so, what is the algorithm used?

**Comment by beorn7** [comment]
Not sure what you mean by "response delay percentile" specifically, but there are various ways of calculating percentiles AKA quantiles in Prometheus:
- First and foremost, see https://prometheus.io/docs/practices/histograms/ to understand the Summary and the (classic) Histogram metric type. They allow you to calculate quantiles of distributions gathered by the instrumented target.
- There is also a [`quantile` aggregation operator](https://prometheus.io/docs/prometheus/latest/querying/operators/#aggregation-operators) to calculate quantiles over dimensions.
- Finally, there is [`quantile_over_time`](https://prometheus.io/docs/prometheus/latest/querying/functions/#aggregation_over_time) to calculate quantiles over time.

Also note the new experimental Native Histograms to increase the accuracy of quantile estimations.
